controllers/argocd/util.go

```diff
@@ -186,27 +186,35 @@ func getArgoContainerImage(cr *argoproj.ArgoCD) string {
 
 // getRepoServerContainerImage will return the container image for the Repo server.
 //
-// There are three possible options for configuring the image, and this is the
+// There are four possible options for configuring the image, and this is the
 // order of preference.
 //
 // 1. from the Spec, the spec.repo field has an image and version to use for
 // generating an image reference.
-// 2. from the Environment, this looks for the `ARGOCD_REPOSERVER_IMAGE` field and uses
+// 2. from the Environment, this looks for the `ARGOCD_IMAGE` field and uses
 // that if the spec is not configured.
-// 3. the default is configured in common.ArgoCDDefaultRepoServerVersion and
-// common.ArgoCDDefaultRepoServerImage.
+// 3. from the Spec, the spec.version field has an image and version to use for
+// generating an image reference
```

Comment:
Based on current logic image value from `.spec.version` will be used over `ARGOCD_IMAGE` env var, if both are set, which is fine as it is the same for other components. Can you switch these statements ?